execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -652,6 +728,28 @@ where
     }
 }
 
+pub fn put_trie<K, V, T, S, E>(
+    _correlation_id: CorrelationId,
+    txn: &mut T,
+    store: &S,
+    trie: &Trie<K, V>,
+) -> Result<(), E>
```

Comment:
We should discuss how we are planning to validate that this thing has the right hash.